sycl/test-e2e/format.py

```diff
@@ -323,19 +357,18 @@ def get_extra_env(sycl_devices):
             test, litConfig, useExternalSh, script, tmpBase
         )
 
-        if len(devices_for_test) > 1 or test.config.test_mode == "build-only":
-            return result
-
-        # Single device - might be an XFAIL.
-        device = devices_for_test[0]
-        if "*" in test.xfails or self.getMatchedFromList(
-            test.config.sycl_dev_features[device], test.xfails
-        ):
-            if result.code is lit.Test.PASS:
-                result.code = lit.Test.XPASS
-            # fail -> expected fail
-            elif result.code is lit.Test.FAIL:
-                result.code = lit.Test.XFAIL
-            return result
-
+        # Single triple/device - might be an XFAIL.
+        def map_result(features, code):
+            if "*" in test.xfails or self.getMatchedFromList(features, test.xfails):
+                if code is lit.Test.PASS:
+                    code = lit.Test.XPASS
+                elif code is lit.Test.FAIL:
+                    code = lit.Test.XFAIL
+            return code
+
+        if len(triples) == 1 and len(devices_for_test) == 0:
```

Comment:
when can `devices_for_test` be empty. is it only for `build-only` mode? if so should we just check for that?